stock_move_location/wizard/stock_move_location.py

```diff
@@ -0,0 +1,332 @@
+        action.update(
+            {"view_mode": "form", "views": [(form_view, "form")], "res_id": pickinig_id}
+        )
+        return action
+
+    def _get_group_quants(self):
+        location_id = self.origin_location_id
+        # Using sql as search_group doesn't support aggregation functions
+        # leading to overhead in queries to DB
+        query = """
+            SELECT product_id, lot_id, SUM(quantity) AS quantity,
+                SUM(reserved_quantity) AS reserved_quantity
+            FROM stock_quant
+            WHERE location_id = %s
+            GROUP BY product_id, lot_id
+        """
+        self.env.cr.execute(query, (location_id.id,))
+        return self.env.cr.dictfetchall()
+
+    def _get_stock_move_location_lines_values(self):
+        product_obj = self.env["product.product"]
+        product_data = []
+        for group in self._get_group_quants():
+            product = product_obj.browse(group.get("product_id")).exists()
+            # Apply the putaway strategy
+            location_dest_id = (
+                self.apply_putaway_strategy
+                and self.destination_location_id._get_putaway_strategy(product).id
+                or self.destination_location_id.id
+            )
+            product_data.append(
+                {
+                    "product_id": product.id,
+                    "move_quantity": group.get("quantity") or 0,
+                    "max_quantity": group.get("quantity") or 0,
+                    "reserved_quantity": group.get("reserved_quantity"),
+                    "origin_location_id": self.origin_location_id.id,
+                    "destination_location_id": location_dest_id,
+                    # cursor returns None instead of False
+                    "lot_id": group.get("lot_id") or False,
+                    "product_uom_id": product.uom_id.id,
+                    "custom": False,
+                }
+            )
+        return product_data
+
+    @api.onchange("origin_location_id")
+    def onchange_origin_location(self):
+        # Get origin_location_disable context key to prevent load all origin
+        # location products when user opens the wizard from stock quants to
+        # move it to other location.
+        if (
+            not self.env.context.get("origin_location_disable")
+            and self.origin_location_id
+        ):
+            lines = []
+            line_model = self.env["wiz.stock.move.location.line"]
+            for line_val in self._get_stock_move_location_lines_values():
+                if line_val.get("max_quantity") <= 0:
+                    continue
+                line = line_model.create(line_val)
+                line.max_quantity = line.get_max_quantity()
+                line.reserved_quantity = line.reserved_quantity
+                lines.append(line)
+            self.update(
+                {"stock_move_location_line_ids": [(6, 0, [line.id for line in lines])]}
+            )
+
```

Comment:
```suggestion
          return {'domain': {'stock_move_location_line_ids': [('id', 'in', [line.id for line in lines])]}}
```

Reply:
I think we should add domain in many2many. If not, when we click add line, all line will be displayed. If user chooses the line that location is different with parent original location id, that will cause error.